comp/core/settings/component.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package settings defines the interface for the component that manage settings that can be changed at runtime
+package settings
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"go.uber.org/fx"
+)
+
+// team: agent-shared-components
+
+// SettingNotFoundError is used to warn about non existing/not registered runtime setting
+type SettingNotFoundError struct {
+	Name string
+}
+
+func (e *SettingNotFoundError) Error() string {
+	return fmt.Sprintf("setting %s not found", e.Name)
+}
+
+// RuntimeSettingResponse is used to communicate settings config
+type RuntimeSettingResponse struct {
+	Description string
+	Hidden      bool
+}
+
+// Component is the component type.
+type Component interface {
+	// RuntimeSettings returns the configurable settings
+	RuntimeSettings() map[string]RuntimeSetting
+	// GetRuntimeSetting returns the value of a runtime configurable setting
+	GetRuntimeSetting(setting string) (interface{}, error)
+	// SetRuntimeSetting changes the value of a runtime configurable setting
+	SetRuntimeSetting(setting string, value interface{}, source model.Source) error
+
+	// API related functions
+	// Todo: (Components) Remove these functions once we can register routes using FX value groups
+	// GetFullConfig returns the full config
+	GetFullConfig(cfg config.Config, namespaces ...string) http.HandlerFunc
+	// GetValue allows to retrive the runtime setting
+	GetValue(setting string, w http.ResponseWriter, r *http.Request)
```

Comment:
Could you provide an example, or just explain, how `GetValue` is intended to be used by a caller? I'm confused that it doesn't return a value, despite being named `Get...`, but rather writes to the http.Response. This indicates to me that either it's misnamed or I'm missing the usecase.

Same with `SetValue`.

Reply:
This functions are meant to use by the API component. This is a middle step until we can register routes using Fx value groups.